src/PointwiseFunctions/Hydro/StressEnergy.cpp

```diff
@@ -70,23 +99,78 @@ void stress_trace(gsl::not_null<Scalar<DataType>*> result,
           (square(get(lorentz_factor)) * get(spatial_velocity_squared) + 1.);
 }
 
+template <typename DataType>
+void stress_energy_tensor(
+    gsl::not_null<tnsr::AA<DataType, 3>*> result,
+    const Scalar<DataType>& rest_mass_density,
+    const Scalar<DataType>& specific_internal_energy,
+    const Scalar<DataType>& pressure, const Scalar<DataType>& lorentz_factor,
+    const Scalar<DataType>& lapse,
+    const Scalar<DataType>& comoving_magnetic_field_magnitude,
+    const tnsr::I<DataType, 3>& spatial_velocity,
+    const tnsr::I<DataType, 3>& shift,
+    const tnsr::I<DataType, 3>& magnetic_field,
+    const tnsr::ii<DataType, 3>& spatial_metric,
+    const tnsr::II<DataType, 3>& inverse_spatial_metric) {
+  const auto inverse_spacetime_metric =
```

Comment:
I'm generally unhappy about the large number of allocations in this function. It would be good to switch to the not_null versions and preallocate in a Variables

Reply:
done.